echopype/echodata/echodata.py

```diff
@@ -315,114 +314,167 @@ def update_platform(
             logger.warning(
-                f"Some variables in the original Platform group will be overwritten: {', '.join(dropped_vars)}"  # noqa
+                f"The following requested variables will not be updated: {', '.join(vars_not_handled)}"  # noqa
             )
-        platform = platform.drop_vars(
-            dropped_vars_target,
-            errors="ignore",
-        )
 
-        def mapping_search_variable(mapping, keys, default=None):
-            for key in keys:
-                if key in mapping:
-                    return mapping[key].data
-            return default
+        vars_notnan_replaced = [
+            platform_var for platform_var, v in mappings_expanded.items() if v[-1]
+        ]
+        if len(vars_notnan_replaced) > 0:
+            logger.warning(
+                f"Some variables with valid data in the original Platform group will be overwritten: {', '.join(vars_notnan_replaced)}"  # noqa
+            )
 
-        num_obs = len(extra_platform_data[time_dim])
-        for platform_var, sourcenames in platform_vars_sourcenames.items():
-            platform = platform.update(
-                {
-                    platform_var: (
-                        "time1",
-                        mapping_search_variable(
-                            extra_platform_data,
-                            sourcenames,
-                            platform.get(platform_var, np.full(num_obs, np.nan)),
-                        ),
-                    )
-                }
+        # Create names for required new time dimensions
+        ext_time_dims = list({v[1] for v in mappings_expanded.values() if v[1] != "scalar"})
+        time_dims_max = max([int(dim[-1]) for dim in platform.dims if dim.startswith("time")])
+        new_time_dims = [f"time{time_dims_max+i+1}" for i in range(len(ext_time_dims))]
+        # Map each new time dim name to the external time dim name:
+        new_time_dims_mappings = {new: ext for new, ext in zip(new_time_dims, ext_time_dims)}
+
+        # Process variable updates by corresponding new time dimensions
+        for time_dim in new_time_dims:
+            ext_time_dim = new_time_dims_mappings[time_dim]
+            mappings_selected = {k: v for k, v in mappings_expanded.items() if v[1] == ext_time_dim}
+            ext_vars = [v[0] for v in mappings_selected.values()]
+            ext_ds = _clip_by_time_dim(extra_platform_data[ext_vars], ext_time_dim)
+
+            # Create new time coordinate and dimension
+            platform = platform.assign_coords(**{time_dim: ext_ds[ext_time_dim].values})
+            # TODO: Update long_name and comment to something more appropriate
+            time_attrs = {
+                "axis": "T",
+                "standard_name": "time",
+                "long_name": "Timestamps for platform motion and orientation data",
+                "comment": "Time coordinate corresponding to platform motion and "
+                "orientation data.",
```

Comment:
```suggestion
                "comment": "Time coordinate corresponding to platform motion and "
                "orientation data originated from a dataset external to the sonar data files.",
```

Reply:
Thank you!! That looks great

Reply:
I don't know what I did that ended up dropping your suggestion for `long_name`. Sorry. I've tried to recreate and will submit it in my next push to this PR.